packages/subgraph/src/utils.ts

```diff
@@ -80,6 +80,15 @@ export function initializeEventEntity(
     entity.set("timestamp", Value.fromBigInt(event.block.timestamp));
     entity.set("transactionHash", Value.fromBytes(event.transaction.hash));
     entity.set("gasPrice", Value.fromBigInt(event.transaction.gasPrice));
+    if (event.receipt) {
+        // @note if we don't cast event.receipt as ethereum.TransactionReceipt,
+        // we get a compile error because it thinks event.receipt is:
+        // ethereum.TransactionReceipt | null
```

Comment:
Normally TS should be able to infer the type with this type of logic:
```ts
    const receipt = event.receipt;
    if (receipt) {
        entity.set("gasUsed", Value.fromBigInt(receipt.gasUsed));
    } else {
        entity.set("gasUsed", Value.fromBigInt(BIG_INT_ZERO));
    }
```

Reply:
yeah I think someone from their side suggested this too, will make this change 👍🏼

Reply:
Remove the comment too.

Reply:
ok